cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -302,43 +307,44 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
   }
 
   @Override
-  public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
+  public void suspendSchedule(ProgramId program, SchedulableProgramType programType, String scheduleName)
     throws ScheduleNotFoundException, SchedulerException {
     String scheduleId = AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName);
     StreamSubscriber subscriber = scheduleSubscribers.get(scheduleId);
     if (subscriber == null) {
-      throw new ScheduleNotFoundException(Id.Schedule.from(program.getApplication(), scheduleName));
+      throw new ScheduleNotFoundException(new ScheduleId(program.getNamespace(), program.getApplication(),
```

Comment:
how come ``ScheduleId`` is passed here, but ``Id.Schedule`` is passed  on line 327? Both should use ``Id.Schedule``.